File: src/Entity/Utilisateur.php
<?php

namespace App\Entity;

use App\Repository\UtilisateurRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups;
use Symfony\Component\Validator\Constraints as Assert;
#[ORM\Entity(repositoryClass: UtilisateurRepository::class)]
class Utilisateur
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    #[Groups("users")]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message:"Le nom est requis")]
    #[Assert\Regex(pattern:"/^[A-Z]+(([',. -][a-zA-Z ])?[a-zA-Z]*)*$/",message:"Le nom format n'est pas valide")]
    #[Groups("users")]
    private ?string $nom = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message:"Le prenom est requis")]
    #[Assert\Regex(pattern:"/^[A-Z]+(([',. -][a-zA-Z ])?[a-zA-Z]*)*$/",message:"Le prenom format n'est pas valide")]
    #[Groups("users")]
    private ?string $prenom = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message:"L'email est requis")]
    #[Assert\Email(message:"L'email format est incorrecte")]
    #[Groups("users")]
    private ?string $email = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message:"Le mot de pass est requis")]
   #[Assert\Length(min:8,minMessage:"Le mot de pass doit ètre suprerieur a 8 caractére")]
   #[Groups("users")] 
   private ?string $mdp = null;
    #[Assert\NotBlank(message:"L'addresse est requis")]
    #[ORM\Column(length: 255)]
    #[Groups("users")]
    private ?string $adresse = null;
    #[Assert\NotBlank(message:"Le numero est requis")]
    #[ORM\Column]
    #[Groups("users")]
    private ?int $num = null;
    //[Assert\NotBlank(message:"Le type est requis")]
    #[ORM\Column(length: 255)]
    #[Groups("users")]
    private ?string $type = null;

    #[ORM\Column(length: 255)]
    #[Groups("users")]
    private ?string $img = null;

    #[ORM\Column(nullable: true)]
    private ?int $etat = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getPrenom(): ?string
    {
        return $this->prenom;
    }

    public function setPrenom(string $prenom): self
    {
        $this->prenom = $prenom;

        return $this;
    }

    public function getEmail(): ?string
    {
        return $this->email;
    }

    public function setEmail(string $email): self
    {
        $this->email = $email;

        return $this;
    }

    public function getMdp(): ?string
    {
        return $this->mdp;
    }

    public function setMdp(string $mdp): self
    {
        $this->mdp = $mdp;

        return $this;
    }

    public function getAdresse(): ?string
    {
        return $this->adresse;
    }

    public function setAdresse(string $adresse): self
    {
        $this->adresse = $adresse;

        return $this;
    }

    public function getNum(): ?int
    {
        return $this->num;
    }

    public function setNum(int $num): self
    {
        $this->num = $num;

        return $this;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): self
    {
        $this->type = $type;

        return $this;
    }

    public function getImg(): ?string
    {
        return $this->img;
    }

    public function setImg(string $img): self
    {
        $this->img = $img;

        return $this;
    }

    public function getEtat(): ?int
    {
        return $this->etat;
    }

    public function setEtat(?int $etat): self
    {
        $this->etat = $etat;

        return $this;
    }
}

File: src/edu/workshopjdbc3a48/entities/Utilisateur.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package edu.workshopjdbc3a48.entities;

/**
 *
 * @author user
 */
public class Utilisateur {
    private int id;
    private String nom;
    private String prenom;
    private String email;
    private String mdp;
    private String Adresse;
    private int num;
    private String type;
    private String img;
    private int etat;
    private String face;
private int code;
    @Override
    public String toString() {
        return "Utilisateur{" + "id=" + id + ", nom=" + nom + ", prenom=" + prenom + ", email=" + email + ", mdp=" + mdp + ", Adresse=" + Adresse + ", num=" + num + ", type=" + type + ", img=" + img + ", etat=" + etat + '}';
    }

    public Utilisateur() {
    }

    public String getFace() {
        return face;
    }

    public Utilisateur(int id, String nom, String prenom, String email, String mdp, String Adresse, int num, String type, String img, int etat, String face) {
        this.id = id;
        this.nom = nom;
        this.prenom = prenom;
        this.email = email;
        this.mdp = mdp;
        this.Adresse = Adresse;
        this.num = num;
        this.type = type;
        this.img = img;
        this.etat = etat;
        this.face = face;
    }

    public Utilisateur(int id, String nom, String prenom, String email, String mdp, String Adresse, int num, String type, String img, int etat, String face, int code) {
        this.id = id;
        this.nom = nom;
        this.prenom = prenom;
        this.email = email;
        this.mdp = mdp;
        this.Adresse = Adresse;
        this.num = num;
        this.type = type;
        this.img = img;
        this.etat = etat;
        this.face = face;
        this.code = code;
    }

    public void setFace(String face) {
        this.face = face;
    }

    public Utilisateur(String nom, String prenom, String email, String mdp, String Adresse, int num, String type, String img, int etat) {
        this.nom = nom;
        this.prenom = prenom;
        this.email = email;
        this.mdp = mdp;
        this.Adresse = Adresse;
        this.num = num;
        this.type = type;
        this.img = img;
        this.etat = etat;
    }

    public Utilisateur(String nom, String prenom, String email, String mdp, String Adresse, int num, String type, String img, int etat, String face) {
        this.nom = nom;
        this.prenom = prenom;
        this.email = email;
        this.mdp = mdp;
        this.Adresse = Adresse;
        this.num = num;
        this.type = type;
        this.img = img;
        this.etat = etat;
        this.face = face;
    }

    public Utilisateur(int id, String nom, String prenom, String email, String mdp, String Adresse, int num, String type, String img, int etat) {
        this.id = id;
        this.nom = nom;
        this.prenom = prenom;
        this.email = email;
        this.mdp = mdp;
        this.Adresse = Adresse;
        this.num = num;
        this.type = type;
        this.img = img;
        this.etat = etat;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public int getCode() {
        return code;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public void setPrenom(String prenom) {
        this.prenom = prenom;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public void setMdp(String mdp) {
        this.mdp = mdp;
    }

    public void setAdresse(String Adresse) {
        this.Adresse = Adresse;
    }

    public void setNum(int num) {
        this.num = num;
    }

    public void setType(String type) {
        this.type = type;
    }

    public void setImg(String img) {
        this.img = img;
    }

    public void setEtat(int etat) {
        this.etat = etat;
    }

    public int getId() {
        return id;
    }

    public String getNom() {
        return nom;
    }

    public String getPrenom() {
        return prenom;
    }

    public String getEmail() {
        return email;
    }

    public String getMdp() {
        return mdp;
    }

    public String getAdresse() {
        return Adresse;
    }

    public int getNum() {
        return num;
    }

    public String getType() {
        return type;
    }

    public String getImg() {
        return img;
    }

    public int getEtat() {
        return etat;
    }
    
}
